### Eclipse Workspace Patch 1.0
#P Uae4All
Index: AndroidManifest.xml
===================================================================
--- AndroidManifest.xml	(revision 3166)
+++ AndroidManifest.xml	(working copy)
@@ -21,10 +21,10 @@
 				android:excludeFromRecents="true" 
         >
             <intent-filter>
-                <!-- 
-                <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
+                <action android:name="android.intent.action.MAIN" />
                 <category android:name="tv.ouya.intent.category.GAME" />
+                <!-- 
                 <action android:name="com.sec.android.airview.HOVER" />
                  -->
                 
Index: src/pandora/uae4all/sdl/Video.java
===================================================================
--- src/pandora/uae4all/sdl/Video.java	(revision 3110)
+++ src/pandora/uae4all/sdl/Video.java	(working copy)
@@ -652,7 +652,9 @@
 		Settings.nativeSetEnv( "DISPLAY_RESOLUTION_WIDTH", String.valueOf(Math.max(mWidth, mHeight)) );
 		Settings.nativeSetEnv( "DISPLAY_RESOLUTION_HEIGHT", String.valueOf(Math.min(mWidth, mHeight)) ); // In Kitkat with immersive mode, getWindowManager().getDefaultDisplay().getMetrics() return inaccurate height
 
-		DemoRenderer.nativeSetSaveDir(context.getIntent().getStringExtra("stateDir"));
+		String saveDir = context.getIntent().getStringExtra("stateDir");
+		if (saveDir==null) saveDir = "/sdcard/";
+		DemoRenderer.nativeSetSaveDir(saveDir);
 		DemoRenderer.nativeSetSaveSlot(0);
 		
 		accelerometer = new AccelerometerReader(context);
@@ -943,6 +945,7 @@
 		});
 		
 		String customConfig = context.getIntent().getStringExtra("conf");
+		if (customConfig == null) customConfig = "/data/data/xtvapps.retrobox.v2/uaeconfig.conf";
 		mRenderer = new DemoRenderer(context, customConfig);
 		setRenderer(mRenderer);
 		
